observability/logging/logging-6.0/log6x-clf.adoc

```diff
@@ -0,0 +1,91 @@
+:_mod-docs-content-type: ASSEMBLY
+include::_attributes/common-attributes.adoc[]
+[id="log6x-CLF_{context}"]
+= Configuring log forwarding
+:context: logging-6x
+
+toc::[]
+
+The `ClusterLogForwarder` CR serves as a single point of configuration for log forwarding, making it easier to manage and maintain log collection and forwarding rules.
+
+* Defines inputs (sources) for log collection
+* Specifies outputs (destinations) for log forwarding
+* Configures filters for log processing
+* Defines pipelines to route logs from inputs to outputs
+* Indicates management state (managed or unmanaged)
+
+// Needs engineering eval re applicability to 6.0
+include::modules/log-forwarding-implementations.adoc[leveloffset=+1]
+
+// This needs to be called out extensively.
+[source,yaml]
+----
+apiVersion: observability.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+spec:
+  inputs:
+    - application
+    - infrastructure
+    - audit
+  outputs:
+    - lokiStack
+    - elasticsearch
+    - splunk
+  filters:
+    - kubeAPIAudit
+    - detectMultilineException
+    - parse
+    - openshiftLabels
+    - drop
+    - prune
+  pipelines:
+    - name: application-logs
+      input: application
+      output: lokiStack
+      filters:
+        - parse
+        - openshiftLabels
+  managementState:
+    managed: true
+----
+
+[id="log6x-CLF-tuning_{context}"]
+== Output tuning
+//need to validate how much of this applies to 6.0 & edit as needed.
+include::modules/logging-delivery-tuning.adoc[leveloffset=+1]
```

Comment:
@anpingli / @jcantrill : Could you help me understand if this module can be used for 6.0 & if not what changes are needed?

Reply:
We have a Jira: [OBSDOCS-1169](https://issues.redhat.com/browse/OBSDOCS-1169) for Output tuning section discussion. I believe this discussion can be covered in https://github.com/openshift/openshift-docs/pull/78477 , which is created for verification of the CLF content. So, that is the reason I have covered [OBSDOCS-1170](https://issues.redhat.com/browse/OBSDOCS-1170) and [OBSDOCS-1169](https://issues.redhat.com/browse/OBSDOCS-1169) in one PR. Please let me know if you have any other thoughts. Thanks!

Cc @jcantrill @anpingli @libander @briandooley

Reply:
Output tuning is functionally the same as in 5.9 though it has moved from common to outputs to each individual spec.  This was done because not all outputs support the same options

Reply:
AtLeastOnce is  atLeastOnce in 6.-0
AtMostOnce is atMostOnce  in 6.0

Reply:
May be we need to update ClusterLogForwarder CR tuning options as follows?

apiVersion: "observability.openshift.io/v1"
kind: ClusterLogForwarder
metadata:
# ...
spec:
    outputs:
        tuning:
           delivery: atLeastOnce 
           maxWrite: <integer> 
           compression: none 
           minRetryDuration: 1s 
           maxRetryDuration: 1s     
# ...

Reply:
apiVersion: "observability.openshift.io/v1"
kind: ClusterLogForwarder
metadata:
# ...
spec:
    outputs:
    - name : <output-name>
      type: <output_type>
      <output_type>
        tuning:
           delivery: atLeastOnce 
           maxWrite: <integer> 
           compression: none 
           minRetryDuration: 1s 
           maxRetryDuration: 1s

Reply:
please refer to https://github.com/openshift/enhancements/blob/master/enhancements/cluster-logging/logs-observability-openshift-io-apis.md#clusterlogforwarer-customresourcedefinition

Reply:
This problem in the current documentation is reflected in documentation bug https://issues.redhat.com/browse/OBSDOCS-1049

Reply:
Tagging @libander so that we can cover this as part of [OBSDOCS-1169](https://issues.redhat.com/browse/OBSDOCS-1169).